internal/resources/slo/resource_slo.go

```diff
@@ -511,6 +529,26 @@ func packQuery(query map[string]interface{}) (slo.SloV00Query, error) {
 		return sloQuery, nil
 	}
 
+	if query["type"] == "grafana_queries" {
+		freeformquery := query["grafana_queries"].([]interface{})[0].(map[string]interface{})
```

Comment:
issue: it appears that you're accessing the `[0]` position in the slice without first checking the length, which could panic.

suggestion: remove `freeform` from the variable name and just use `queryObject` for the outer object here.

question: is this unchecked type assertion for the `["grafana_queries"].(string)` consuming the terraform resource where the value is actually a string? I ask because the API type if structured and I was surprised so see the `.(string)`

Maybe I could could get a live walkthrough of this later this week if I don't find the time to play with the terraform-provider locally?